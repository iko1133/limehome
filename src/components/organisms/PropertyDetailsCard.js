import React from "react";
import styled from "styled-components/native";

import Divider from "atoms/Divider";
import IconText from "atoms/IconText";

const PropertyDetailsViewContainer = styled.TouchableOpacity`
  background-color: ${({ theme }) => theme.BACKGROUND_COLOR};
  border: 1px solid ${({ theme }) => theme.DEFAULT_BORDER_COLOR};
  border-radius: 4px;
  bottom: 40px;
  flex-direction: row;
  overflow: hidden;
  width: 90%;
  position: absolute;
`;

const PropertyImageContainer = styled.View`
  aspect-ratio: 1;
  height: 100%;
`;

const PropertyImage = styled.Image`
  height: 100%;
  width: 100%;
`;

const PropertyRatingContainer = styled.View`
  background-color: ${({ theme }) => theme.BACKGROUND_COLOR};
  border-radius: 6px;
  right: 8px;
  top: 10px;
  padding: 4px 5px;
  position: absolute;
`;

const PropertyDescriptionContainer = styled.View`
  flex: 1;
  padding: 12px 12px;
`;

const PropertyName = styled.Text`
  font-weight: bold;
  font-size: 18px;
  margin-bottom: 8px;
`;

const PropertyLocationText = styled(IconText)``;

const PropertyPriceFromLabel = styled.Text``;

const PropertyPriceText = styled.Text`
  color: ${({ theme }) => theme.SPECIAL_TEXT_COLOR};
`;

/**
 * A card showing a little bit of information of the property given
 * @param {object} propertyDetails - an object containing the information of the property displayed by the component
 * @param {function} onPress - a callback that's called when the card is pressed
 */

const PropertyDetailsCard = ({ propertyDetails, onPress }) => {
  return (
    <PropertyDetailsViewContainer onPress={onPress} activeOpacity={0.8}>
      <PropertyImageContainer>
        <PropertyImage
          source={{
            uri: propertyDetails.images[0]?.url,
          }}
        />

        <PropertyRatingContainer>
          <PropertyLocationText
            text="4.5"
            iconName="star"
            iconSize={18}
            iconLeft={false}
          />
        </PropertyRatingContainer>
      </PropertyImageContainer>

      <PropertyDescriptionContainer>
        <PropertyName>{propertyDetails.name}</PropertyName>

        <PropertyLocationText
          text={`${Math.round(propertyDetails.distance)} km from city center`}
          iconName="location-pin"
        />

        <Divider />

        <PropertyPriceFromLabel>
          From{" "}
          <PropertyPriceText>
            {(propertyDetails.lowest_price_per_night ?? 0).toFixed(2)}
            â‚¬/Night
          </PropertyPriceText>
        </PropertyPriceFromLabel>
      </PropertyDescriptionContainer>
    </PropertyDetailsViewContainer>
  );
};

export default PropertyDetailsCard;
