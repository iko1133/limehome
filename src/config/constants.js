export const MAP_DEFAULT_COORDINATES = {
  latitude: 52.520008,
  longitude: 13.404954,
  latitudeDelta: 0.15,
  longitudeDelta: 0.15,
};
