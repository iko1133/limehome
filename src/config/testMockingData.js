export const PROPERTIES = [
  {
    id: 99,
    name: "berlin schlüterstraße",
    city: "berlin",
    city_id: 32,
    street: "schlüterstraße",
    location: {
      lat: 52.50176279999999,
      lng: 13.3164893,
      city: "Berlin",
    },
    distance: 6.321157219990815,
    images: [
      {
        url: "https://limehome.imgix.net/properties/99/9fee5516-9d41-41f7-95ea-6ac72c49fb86.jpg",
        is_portrait: false,
        position: 0,
        unit_group_ids: [16],
        tags: [],
      },
    ],
    lowest_price_per_night: null,
    lowest_price_per_month: null,
  },
  {
    id: 152,
    name: "berlin werftstraße",
    city: "berlin",
    city_id: 32,
    street: "werftstraße ",
    location: {
      lat: 52.52259309999999,
      lng: 13.3589213,
      city: "Berlin",
    },
    distance: 3.127749150912429,
    images: [
      {
        url: "https://limehome.imgix.net/properties/152/bff791de-57fa-450b-869d-0a7bfac7a55c.jpg",
        is_portrait: false,
        position: 0,
        unit_group_ids: [178],
        tags: [],
      },
    ],
    lowest_price_per_night: null,
    lowest_price_per_month: null,
  },
];

export const FIXED_PROPERTIES = [
  {
    id: 99,
    name: "berlin schlüterstraße",
    city: "berlin",
    city_id: 32,
    street: "schlüterstraße",
    location: {
      latitude: 52.50176279999999,
      longitude: 13.3164893,
      city: "Berlin",
    },
    distance: 6.321157219990815,
    images: [
      {
        url: "https://limehome.imgix.net/properties/99/9fee5516-9d41-41f7-95ea-6ac72c49fb86.jpg",
        is_portrait: false,
        position: 0,
        unit_group_ids: [16],
        tags: [],
      },
    ],
    lowest_price_per_night: null,
    lowest_price_per_month: null,
  },
  {
    id: 152,
    name: "berlin werftstraße",
    city: "berlin",
    city_id: 32,
    street: "werftstraße ",
    location: {
      latitude: 52.52259309999999,
      longitude: 13.3589213,
      city: "Berlin",
    },
    distance: 3.127749150912429,
    images: [
      {
        url: "https://limehome.imgix.net/properties/152/bff791de-57fa-450b-869d-0a7bfac7a55c.jpg",
        is_portrait: false,
        position: 0,
        unit_group_ids: [178],
        tags: [],
      },
    ],
    lowest_price_per_night: null,
    lowest_price_per_month: null,
  },
];
