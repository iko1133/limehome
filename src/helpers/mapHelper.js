// Copies given object, except, switches names of the location property's properties for the
// marker objects to function properly
// lgn to longitude
// lat to latitude
export const fixObjectLocationProperty = (object) => ({
  ...object,
  location: {
    ...object.location,
    longitude: object.location?.lng,
    latitude: object.location?.lat,
  },
});
