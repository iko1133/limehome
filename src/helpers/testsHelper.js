import { ThemeProvider } from "styled-components/native";

import { lightTheme } from "styles/theme";

/**
 * Wraps given children into a ThemeProvider with lightTheme set as the main theme
 * @param {object} children - children to be rendered inside ThemeProvider
 */
export const WrapInThemeProvider = (children) => (
  <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
);
