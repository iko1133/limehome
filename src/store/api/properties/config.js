// list of urls needed
const config = {
  propertiesList: "properties/v1/public/properties/",
  propertyDetails: "properties/v1/public/properties/",
};

export default config;
