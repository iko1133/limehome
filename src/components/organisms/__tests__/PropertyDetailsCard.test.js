import React from "react";
import renderer from "react-test-renderer";

import { WrapInThemeProvider } from "helpers/testsHelper";
import { FIXED_PROPERTIES } from "config/testMockingData";
import PropertyDetailsCard from "../PropertyDetailsCard";

// Fixing vector icons problem with jest
jest.mock(
  "@expo/vector-icons/build/vendor/react-native-vector-icons/lib/create-icon-set.js",
  () => {
    return () => "";
  }
);

describe("Testing PropertyDetailsCard", () => {
  it("Renders property details card with property 0 on it", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(
          <PropertyDetailsCard propertyDetails={FIXED_PROPERTIES[0]} />
        )
      )
    ).toMatchSnapshot());

  it("Renders property details card with property 1 on it", () =>
    expect(
      renderer.create(
        WrapInThemeProvider(
          <PropertyDetailsCard propertyDetails={FIXED_PROPERTIES[1]} />
        )
      )
    ).toMatchSnapshot());
});
