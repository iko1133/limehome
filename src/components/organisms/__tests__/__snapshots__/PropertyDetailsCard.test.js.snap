// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PropertyDetailsCard Renders property details card with property 0 on it 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#F7F1E9",
      "borderColor": "#9D9E9F",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "bottom": 40,
      "flexDirection": "row",
      "opacity": 1,
      "overflow": "hidden",
      "position": "absolute",
      "width": "90%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "aspectRatio": 1,
          "height": "100%",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://limehome.imgix.net/properties/99/9fee5516-9d41-41f7-95ea-6ac72c49fb86.jpg",
        }
      }
      style={
        Array [
          Object {
            "height": "100%",
            "width": "100%",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F7F1E9",
            "borderRadius": 6,
            "paddingBottom": 4,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 4,
            "position": "absolute",
            "right": 8,
            "top": 10,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {},
            ]
          }
        >
          4.5
        </Text>
        <Text />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 8,
          },
        ]
      }
    >
      berlin schlüterstraße
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {},
        ]
      }
    >
      <Text />
      <Text
        style={
          Array [
            Object {},
          ]
        }
      >
        6 km from city center
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "darkgray",
            "height": 1,
            "marginBottom": 8,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 8,
            "width": "100%",
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {},
        ]
      }
    >
      From
       
      <Text
        style={
          Array [
            Object {
              "color": "#B26422",
            },
          ]
        }
      >
        0.00
        €/Night
      </Text>
    </Text>
  </View>
</View>
`;

exports[`Testing PropertyDetailsCard Renders property details card with property 1 on it 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#F7F1E9",
      "borderColor": "#9D9E9F",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "bottom": 40,
      "flexDirection": "row",
      "opacity": 1,
      "overflow": "hidden",
      "position": "absolute",
      "width": "90%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "aspectRatio": 1,
          "height": "100%",
        },
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://limehome.imgix.net/properties/152/bff791de-57fa-450b-869d-0a7bfac7a55c.jpg",
        }
      }
      style={
        Array [
          Object {
            "height": "100%",
            "width": "100%",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F7F1E9",
            "borderRadius": 6,
            "paddingBottom": 4,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 4,
            "position": "absolute",
            "right": 8,
            "top": 10,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {},
            ]
          }
        >
          4.5
        </Text>
        <Text />
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "paddingBottom": 12,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 12,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 8,
          },
        ]
      }
    >
      berlin werftstraße
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {},
        ]
      }
    >
      <Text />
      <Text
        style={
          Array [
            Object {},
          ]
        }
      >
        3 km from city center
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "darkgray",
            "height": 1,
            "marginBottom": 8,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 8,
            "width": "100%",
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {},
        ]
      }
    >
      From
       
      <Text
        style={
          Array [
            Object {
              "color": "#B26422",
            },
          ]
        }
      >
        0.00
        €/Night
      </Text>
    </Text>
  </View>
</View>
`;
